extensions/2.0/Vendor/EXT_structural_metadata/README.md

```diff
@@ -11,40 +13,522 @@
+
+A property texture may provide channels for only a subset of the properties of its class, but class properties marked `required: true` must not be omitted.
+
+Several constraints are imposed to ensure compatibility of texture storage with various property types:
+
+* Variable-length arrays are not supported.
+* Data type and bit depth of the image must be compatible with the property type.
+
+Enum values may be encoded in images, as integer values according to their enum value type (see [Enum](#enum)).
+
+> **Implementation note:** Use of floating-point properties in a property texture would require a floating point-compatible image format like KTX2, provided by an additional extension.
+
+> **Example:** Property texture implementing a "wall" class, with property values stored in a glTF texture at index 0 and indexed by `TEXCOORD_0`. Each property of the `"wall"` class is stored in one channel of the texture.
+>
+> ![Property Texture](figures/property-texture.png)
+>
+> _Class and property texture_
+>
+> ```jsonc
+> {
+>   "extensions": {
+>     "EXT_structural_metadata": {
+>       "schema": {
+>         "classes": {
+>           "wall": {
+>             "name": "Wall Temperature vs. Insulation",
+>             "properties": {
+>               "insideTemperature": {
+>                 "name": "Inside Temperature",
+>                 "componentType": "UINT8"
+>               },
+>               "outsideTemperature": {
+>                 "name": "Outside Temperature",
+>                 "componentType": "UINT8"
+>               },
+>               "insulation": {
+>                 "name": "Insulation Thickness",
+>                 "componentType": "UINT8",
+>                 "normalized": true
+>               }
+>             }
+>           }
+>         }
+>       },
+>       "propertyTextures": [
+>         {
+>           "class": "wall",
+>           "index": 0,
+>           "texCoord": 0,
+>           "properties": {
+>             "insideTemperature": {
+>               "channels": [0]
+>             },
+>             "outsideTemperature": {
+>               "channels": [1]
+>             },
+>             "insulation": {
+>               "channels": [2]
+>             }
+>           }
+>         }
+>       ]
+>     }
+>   }
+> }
+> ```
+>
+> _Primitive_
+>
+> ```jsonc
+> {
+>   "primitives": [
+>     {
+>       "attributes": {
+>         "POSITION": 0,
+>         "TEXCOORD_0": 1
+>       },
+>       "indices": 2,
+>       "material": 0,
+>       "extensions": {
+>         "EXT_structural_metadata": {
+>           "propertyTextures": [0]
+>         }
+>       }
+>     }
+>   ]
+> }
+> ```
+
+A `propertyTexture` object extends the glTF [`textureInfo`](../../../../specification/2.0/schema/textureInfo.schema.json) object. `texCoord` specifies a texture coordinate set in the referring primitive.
```

Comment:
This behavior changed between `EXT_mesh_features` and `EXT_structural_metadata`. Now each individual property extends the `textureInfo` object. This way the number of class properties is no longer constrained by the number of texture channels. The new structure looks like this:

```json
      "propertyTextures": [
        {
          "class": "wall",
          "properties": {
            "insideTemperature": {
              "index": 0,
              "texCoord": 0,
              "channels": [0]
            },
            "outsideTemperature": {
              "index": 0,
              "texCoord": 0,
              "channels": [1]
            },
            "insulation": {
              "index": 0,
              "texCoord": 0,
              "channels": [2]
            }
          }
        }
      ]
```